tls/s2n_record_write.c

```diff
@@ -482,8 +512,6 @@ int s2n_record_writev(struct s2n_connection *conn, uint8_t content_type, const s
         conn->server = current_server_crypto;
     }
 
-    conn->wire_bytes_out += actual_fragment_length + S2N_TLS_RECORD_HEADER_LENGTH;
```

Comment:
Doesn't this still need to happen somewhere?

Reply:
I saw this was also being written to in [s2n_flush](https://github.com/aws/s2n-tls/blob/3271f9503c8ef0a9221a0f05bd7f647e9e2124db/tls/s2n_send.c#L52). 

This seemed wrong since it means we are counting the bytes we write twice. Once when we create the record and again when we write data over the socket in `s2n_flush`.

This means [s2n_connection_get_wire_bytes_out](https://github.com/aws/s2n-tls/blob/3271f9503c8ef0a9221a0f05bd7f647e9e2124db/api/s2n.h#L2673) is over-reporting how many bytes are being sent over the wire.

I removed this line if favor of `s2n_flush` recording how many bytes were successfully sent over the socket.

Reply:
makes sense